{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "pd.options.mode.chained_assignment = None  # default='warn', Mutes warnings when copying a slice from a DataFrame.\n",
    "\n",
    "DATA_FOLDER = './Data/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average* per year of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Initialization\n",
    "#The dataFrame we gonna fill and work with.\n",
    "df = pd.DataFrame()\n",
    "EBOLA_FOLDER = DATA_FOLDER + \"Ebola/\"\n",
    "COLUMNS_TO_KEEP = pd.DataFrame([{'guinea': ['Date', 'Description' , 'Totals'],\n",
    "                    'liberia':['Date', 'Variable', 'National'],\n",
    "                    'sl':['date', 'variable', 'National']}])\n",
    "\n",
    "GOOD_HEADER = ['Date', 'Description' , 'Totals']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Country</th>\n",
       "      <th>Date</th>\n",
       "      <th>Description</th>\n",
       "      <th>Totals</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New cases of suspects</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New cases of probables</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New cases of confirmed</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total new cases registered so far</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total cases of suspects</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total cases of probables</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total cases of confirmed</td>\n",
       "      <td>351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Cumulative (confirmed + probable + suspects)</td>\n",
       "      <td>495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New deaths registered today</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New deaths registered today (confirmed)</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New deaths registered today (probables)</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>New deaths registered today (suspects)</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total deaths of suspects</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total deaths of probables</td>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total deaths of confirmed</td>\n",
       "      <td>228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total deaths (confirmed + probables + suspects)</td>\n",
       "      <td>363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total PEC center today</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total PEC center today (confirmed)</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total PEC center today (probables)</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total PEC center today (suspects)</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total of deaths in confirmed cases in CTE</td>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total of cured in confirmed cases in CTE</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Number of male confirmed cases</td>\n",
       "      <td>161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Number of female confirmed cases</td>\n",
       "      <td>190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Number of male probables cases</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Number of female probables cases</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Number of male suspects cases</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Number of female suspects cases</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total number of male cases</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>guinea</td>\n",
       "      <td>2014-08-04</td>\n",
       "      <td>Total number of female cases</td>\n",
       "      <td>277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7098</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>cum_probable</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7099</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>cum_confirmed</td>\n",
       "      <td>6638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7100</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>death_suspected</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7101</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>death_probable</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7102</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>death_confirmed</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7103</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>cfr</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7104</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>etc_new_admission</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7105</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>etc_currently_admitted</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7106</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>etc_cum_admission</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7107</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>etc_new_deaths</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7108</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>etc_cum_deaths</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7109</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>etc_new_discharges</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7110</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>etc_cum_discharges</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7111</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>cum_contacts</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7112</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>cum_completed_contacts</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7113</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>contacts_followed</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7114</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>new_contacts</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7115</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>contacts_healthy</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7116</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>contacts_ill</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7117</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>contacts_not_seen</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7118</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>new_completed_contacts</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7119</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>percent_seen</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7120</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>total_lab_samples</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7121</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>repeat_samples</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7122</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>new_samples</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7123</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>new_positive</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7124</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>new_negative</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7125</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>pending</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7126</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>positive_corpse</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7127</th>\n",
       "      <td>sl</td>\n",
       "      <td>2014-12-13</td>\n",
       "      <td>negative_corpse</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7128 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Country        Date                                      Description  \\\n",
       "0     guinea  2014-08-04                            New cases of suspects   \n",
       "1     guinea  2014-08-04                           New cases of probables   \n",
       "2     guinea  2014-08-04                           New cases of confirmed   \n",
       "3     guinea  2014-08-04                Total new cases registered so far   \n",
       "4     guinea  2014-08-04                          Total cases of suspects   \n",
       "5     guinea  2014-08-04                         Total cases of probables   \n",
       "6     guinea  2014-08-04                         Total cases of confirmed   \n",
       "7     guinea  2014-08-04     Cumulative (confirmed + probable + suspects)   \n",
       "8     guinea  2014-08-04                      New deaths registered today   \n",
       "9     guinea  2014-08-04          New deaths registered today (confirmed)   \n",
       "10    guinea  2014-08-04          New deaths registered today (probables)   \n",
       "11    guinea  2014-08-04           New deaths registered today (suspects)   \n",
       "12    guinea  2014-08-04                         Total deaths of suspects   \n",
       "13    guinea  2014-08-04                        Total deaths of probables   \n",
       "14    guinea  2014-08-04                        Total deaths of confirmed   \n",
       "15    guinea  2014-08-04  Total deaths (confirmed + probables + suspects)   \n",
       "16    guinea  2014-08-04                           Total PEC center today   \n",
       "17    guinea  2014-08-04               Total PEC center today (confirmed)   \n",
       "18    guinea  2014-08-04               Total PEC center today (probables)   \n",
       "19    guinea  2014-08-04                Total PEC center today (suspects)   \n",
       "20    guinea  2014-08-04        Total of deaths in confirmed cases in CTE   \n",
       "21    guinea  2014-08-04         Total of cured in confirmed cases in CTE   \n",
       "22    guinea  2014-08-04                   Number of male confirmed cases   \n",
       "23    guinea  2014-08-04                 Number of female confirmed cases   \n",
       "24    guinea  2014-08-04                   Number of male probables cases   \n",
       "25    guinea  2014-08-04                 Number of female probables cases   \n",
       "26    guinea  2014-08-04                    Number of male suspects cases   \n",
       "27    guinea  2014-08-04                  Number of female suspects cases   \n",
       "28    guinea  2014-08-04                       Total number of male cases   \n",
       "29    guinea  2014-08-04                     Total number of female cases   \n",
       "...      ...         ...                                              ...   \n",
       "7098      sl  2014-12-13                                     cum_probable   \n",
       "7099      sl  2014-12-13                                    cum_confirmed   \n",
       "7100      sl  2014-12-13                                  death_suspected   \n",
       "7101      sl  2014-12-13                                   death_probable   \n",
       "7102      sl  2014-12-13                                  death_confirmed   \n",
       "7103      sl  2014-12-13                                              cfr   \n",
       "7104      sl  2014-12-13                                etc_new_admission   \n",
       "7105      sl  2014-12-13                           etc_currently_admitted   \n",
       "7106      sl  2014-12-13                                etc_cum_admission   \n",
       "7107      sl  2014-12-13                                   etc_new_deaths   \n",
       "7108      sl  2014-12-13                                   etc_cum_deaths   \n",
       "7109      sl  2014-12-13                               etc_new_discharges   \n",
       "7110      sl  2014-12-13                               etc_cum_discharges   \n",
       "7111      sl  2014-12-13                                     cum_contacts   \n",
       "7112      sl  2014-12-13                           cum_completed_contacts   \n",
       "7113      sl  2014-12-13                                contacts_followed   \n",
       "7114      sl  2014-12-13                                     new_contacts   \n",
       "7115      sl  2014-12-13                                 contacts_healthy   \n",
       "7116      sl  2014-12-13                                     contacts_ill   \n",
       "7117      sl  2014-12-13                                contacts_not_seen   \n",
       "7118      sl  2014-12-13                           new_completed_contacts   \n",
       "7119      sl  2014-12-13                                     percent_seen   \n",
       "7120      sl  2014-12-13                                total_lab_samples   \n",
       "7121      sl  2014-12-13                                   repeat_samples   \n",
       "7122      sl  2014-12-13                                      new_samples   \n",
       "7123      sl  2014-12-13                                     new_positive   \n",
       "7124      sl  2014-12-13                                     new_negative   \n",
       "7125      sl  2014-12-13                                          pending   \n",
       "7126      sl  2014-12-13                                  positive_corpse   \n",
       "7127      sl  2014-12-13                                  negative_corpse   \n",
       "\n",
       "     Totals  \n",
       "0         5  \n",
       "1         0  \n",
       "2         4  \n",
       "3         9  \n",
       "4        11  \n",
       "5       133  \n",
       "6       351  \n",
       "7       495  \n",
       "8         2  \n",
       "9         2  \n",
       "10        0  \n",
       "11        0  \n",
       "12        2  \n",
       "13      133  \n",
       "14      228  \n",
       "15      363  \n",
       "16       31  \n",
       "17       18  \n",
       "18        0  \n",
       "19       13  \n",
       "20      138  \n",
       "21       89  \n",
       "22      161  \n",
       "23      190  \n",
       "24       52  \n",
       "25       81  \n",
       "26        7  \n",
       "27        6  \n",
       "28      220  \n",
       "29      277  \n",
       "...     ...  \n",
       "7098     79  \n",
       "7099   6638  \n",
       "7100    NaN  \n",
       "7101    NaN  \n",
       "7102    NaN  \n",
       "7103    NaN  \n",
       "7104    NaN  \n",
       "7105    NaN  \n",
       "7106    NaN  \n",
       "7107    NaN  \n",
       "7108    NaN  \n",
       "7109    NaN  \n",
       "7110    NaN  \n",
       "7111    NaN  \n",
       "7112    NaN  \n",
       "7113    NaN  \n",
       "7114    NaN  \n",
       "7115    NaN  \n",
       "7116    NaN  \n",
       "7117    NaN  \n",
       "7118    NaN  \n",
       "7119    NaN  \n",
       "7120    NaN  \n",
       "7121    NaN  \n",
       "7122    NaN  \n",
       "7123    NaN  \n",
       "7124    NaN  \n",
       "7125    NaN  \n",
       "7126    NaN  \n",
       "7127    NaN  \n",
       "\n",
       "[7128 rows x 4 columns]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "#For all the directory in Ebola directory load the .csv file in it.\n",
    "for dir in os.listdir(EBOLA_FOLDER):\n",
    "    if \"_data\" in dir:\n",
    "        #Substract _data to only get the country name\n",
    "        country = dir[:-5]\n",
    "        #For each file in the country directory\n",
    "        for file in os.listdir(EBOLA_FOLDER + dir):\n",
    "            if file.endswith(\".csv\"):\n",
    "                tmp = pd.read_csv(EBOLA_FOLDER + dir + '/' + file, usecols= COLUMNS_TO_KEEP[country].at[0])\n",
    "                tmp.columns = GOOD_HEADER\n",
    "                nbRow = tmp.shape[0]\n",
    "                charlist = [country]*nbRow\n",
    "                chararray = np.array(charlist)\n",
    "                #Add a column in the 1st position representing the country.\n",
    "                tmp.insert(0, 'Country', chararray)\n",
    "                if(df.empty):\n",
    "                    df = tmp\n",
    "                else:\n",
    "                    df = df.merge(tmp,how='outer')\n",
    "\n",
    "\n",
    "#Mean of new cases per month & country yearly.\n",
    "\n",
    "\n",
    "df\n",
    "#df.to_csv('df.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'/titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
